.circleci/config.yml

```diff
@@ -0,0 +1,42 @@
+version: 2.1
+
+# reusable environment for all jobs
+executors:
+  main-executor:
+    working_directory: ~/widgets
+    docker:
+      - image: circleci/node:12.7.0-browsers
+
+# reusable command for all jobs
+commands:
+  checkout_from_workspace:
+    description: 'To checkout and attach the workspace'
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/widgets
+
+# define jobs
+jobs:
+  install:
+    executor: main-executor
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - dependency-cache-{{ checksum "package-lock.json" }}
+      - run:
+          name: Install Dependencies
+          command: npm i
```

Comment:
Didn't the team discover some issues with this type of setup and moved over to `npm ci` last week?

Reply:
not sure if `npm ci` works with our logic here.  we don't want to remove the modules if the `package-lock.json` had not been updated. therefore `npm i` is what we need.